hedera-node/src/main/java/com/hedera/services/store/tokens/unique/UniqueTokenStore.java

```diff
@@ -144,6 +145,16 @@ public MerkleUniqueToken get(final NftID id) {
 		return uniqueTokenSupplier.get().get(fromNftID(id));
 	}
 
+	@Override
+	public ResponseCodeEnum adjustBalance(AccountID senderAId, AccountID receiverAId, TokenID tId, long serialNumber) {
+		var nftId = NftID.newBuilder().setTokenID(tId).setSerialNumber(serialNumber).build();
+		if (nftExists(nftId)) {
+			get(nftId).setOwner(EntityId.fromGrpcAccountId(receiverAId));
```

Comment:
- if `nft` does not exist we must return `INVALID_NFT_ID` and not `FAIL_INVALID`
- We should first do the `sanityChecked` part and do the verifications and only after that change the owner of the NFT

Reply:
Confirmed and resolved